yt_dlp/extractor/nytimes.py

```diff
@@ -125,138 +91,427 @@ class NYTimesIE(NYTimesBaseIE):
+        'info_dict': {
+            'id': '1017817',
+            'ext': 'mp4',
+            'title': 'Cranberry Curd Tart',
+            'description': 'md5:ad77a3fc321db636256d4343c5742152',
+            'timestamp': 1447804800,
+            'upload_date': '20151118',
+            'creator': 'David Tanis',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+
+        },
+    }, {
+        'url': 'https://cooking.nytimes.com/recipes/1024781-neapolitan-checkerboard-cookies',
+        'md5': '58df35998241dcf0620e99e646331b42',
+        'info_dict': {
+            'id': '1024781',
+            'ext': 'mp4',
+            'title': 'Neapolitan Checkerboard Cookies',
+            'description': 'md5:ba12394c585ababea951cb6d2fcc6631',
+            'timestamp': 1701302400,
+            'upload_date': '20231130',
+            'creator': 'Sue Li',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+        },
+    }, {
+        'url': 'https://cooking.nytimes.com/recipes/1019516-overnight-oats',
+        'md5': '2fe7965a3adc899913b8e25ada360823',
+        'info_dict': {
+            'id': '1019516',
+            'ext': 'mp4',
+            'timestamp': 1546387200,
+            'description': 'md5:8856ce10239161bd2596ac335b9f9bfb',
+            'upload_date': '20190102',
+            'title': 'Overnight Oats',
+            'creator': 'Genevieve Ko',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+        }
+    }]
+
     def _real_extract(self, url):
         page_id = self._match_id(url)
-
         webpage = self._download_webpage(url, page_id)
 
-        video_id = self._search_regex(
-            r'data-videoid=["\'](\d+)', webpage, 'video id',
-            default=None, fatal=False)
-        if video_id is not None:
-            return self._extract_video_from_id(video_id)
+        json_obj = traverse_obj(
+            self._search_nextjs_data(webpage, page_id), ('props', 'pageProps'), default={})
 
-        podcast_data = self._search_regex(
-            (r'NYTD\.FlexTypes\.push\s*\(\s*({.+?})\s*\)\s*;\s*</script',
-             r'NYTD\.FlexTypes\.push\s*\(\s*({.+})\s*\)\s*;'),
-            webpage, 'podcast data')
-        return self._extract_podcast_from_json(podcast_data, page_id, webpage)
+        info = traverse_obj(json_obj, {
+            'id': ('recipe', 'id', {str_or_none}),
+            'title': ('recipe', 'title'),
+            'description': ('recipe', 'topnote', {clean_html}),
+            'timestamp': ('recipe', 'publishedAt'),
+            'creator': ('recipe', 'contentAttribution', 'cardByline'),
+            'upload_date': ('meta', 'jsonLD', 'video', 'uploadDate', {unified_strdate}),
+            'formats': ('recipe', 'videoSrc', {url_or_none}),
+        })
 
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+            info.get('formats'), info.get('id'), 'mp4', entry_protocol='m3u8_native', m3u8_id='hls')
```

Comment:
`entry_protocol='m3u8_native'` is the default

just use `page_id` matched from the url for the video id -- it looks like it's the same as what's found in the next js data

and again, the `_extract_m3u8_formats` call here is fatal so it's fine to risk a KeyError too
```suggestion
            next_data['recipe']['videoSrc'], page_id, 'mp4', m3u8_id='hls')
```